### ISSUE TITLE
Arithmetic operations lose type data 

### ISSUE THREAD

**Issue by OddBloke** [RustPython/RustPython#92]
This code:

```py
x = 1
print(type(x))
print(type(x-1))
```

produces this output:

```
<class 'int'> 
<class 'type'>
```

This is caused by `VirtualMachine._sub` (and its compatriots) using `self.get_type()` as the type of the new object, rather than using the type information from the arguments.

**Comment by windelbouwman** [comment]
This should be fixed by in PR #99

**Comment by OddBloke** [comment]
This looks to be fixed as expected.
